primitives/api/proc-macro/src/impl_runtime_apis.rs

```diff
@@ -297,11 +297,18 @@ fn generate_runtime_api_base_structures() -> Result<TokenStream> {
 				#crate_::StorageChanges<C::StateBackend, Block>,
 				String
 			> where Self: Sized {
+				let at = #crate_::BlockId::Hash(parent_hash.clone());
+				let state_hash = self.call
+					.runtime_version_at(&at)
+					.map(|v| v.state_version())
```

Comment:
There is https://github.com/paritytech/substrate/blob/630422d6108cbaaca893ab213dde69f3bdaa1f6b/client/executor/src/wasm_runtime.rs#L157.
But I did not really check it is in use (shoud be cached already since it is post execution).